documentation/wallet.adoc

```diff
@@ -0,0 +1,853 @@
+
+[plantuml, format="svg", id="sign-message", width="100%", role="center-image", options="interactive"]
+----
+@startuml
+
+skinparam FontName Poppins
+
+skinparam sequence {
+    ArrowColor #69679D
+    LifeLineBorderColor #69679D
+    LifeLineBackgroundColor #F5F5FF
+
+    ParticipantBorderColor #69679D
+    ParticipantBackgroundColor #F5F5FF
+    ParticipantFontName Poppins
+    ParticipantFontSize 16
+    ParticipantFontColor #000000
+}
+
+participant "User" as user
+participant "dApp" as dapp
+participant "Massa Station" as station
+participant "Massa Wallet API" as wallet_api
+participant "Massa Wallet Core" as wallet_core
+participant "Massa Wallet Embedded Front" as wallet_embedded_front
+
+user -> dapp ++ : Click "Sign In"
+    dapp -> station ++: Request //Sign Message//
+        station -> wallet_api ++: Request //Sign Message//
+        wallet_api -> wallet_api: Execute //Validate Arguments//
+            wallet_api -> wallet_core ++: Execute //Sign Message//
+            wallet_core -> wallet_core: Fetch //Account Data//
+                wallet_core -> wallet_embedded_front ++: Fetch //Password Input//
+                wallet_embedded_front <-> user: Enter password
+                return password
+            wallet_core -> wallet_core: Execute //Unseal Private Key//
+            wallet_core -> wallet_core: Execute //Generate message signature//
+            return message signature
+        return message signature
+    return message signature
+dapp -> dapp: Verify signature
+dapp -> dapp: Open user access
+return logout
+@enduml
+----
+
+===== Backup Account
+The following diagram illustrates the sequence of events when backing up an account.
+
+[plantuml, format="svg", id="backup-account", width="100%", role="center-image", options="interactive"]
+----
+@startuml
+
+skinparam FontName Poppins
+
+skinparam sequence {
+    ArrowColor #69679D
+    LifeLineBorderColor #69679D
+    LifeLineBackgroundColor #F5F5FF
+
+    ParticipantBorderColor #69679D
+    ParticipantBackgroundColor #F5F5FF
+    ParticipantFontName Poppins
+    ParticipantFontSize 16
+    ParticipantFontColor #000000
+}
+
+participant "User" as user
+participant "Massa Wallet Front" as wallet_front
+participant "Massa Wallet API" as wallet_api
+participant "Massa Wallet Core" as wallet_core
+participant "Massa Wallet Embedded Front" as wallet_embedded_front
+
+user -> wallet_front ++ : Click "Backup Account"
+    wallet_front -> wallet_api ++: Request //Backup Account//
+        wallet_api -> wallet_core ++: Execute //Backup Account//
+            wallet_core -> wallet_embedded_front ++: Fetch //Password Input//
+            wallet_embedded_front <-> user: Enter password
+            return password
+        wallet_core -> wallet_core: Execute //Unseal Private Key//
+        wallet_core -> wallet_embedded_front: Execute //Display Private Key//
+        wallet_embedded_front <-> user: Read/Copy Private Key
+        user -> wallet_embedded_front: Close
```

Comment:
What is close ? is it a function to call ? Is this function significant to be listed and described ?

Reply:
the user closes the embedded front (wails popup)

Reply:
This is a user action and not functional concept to be described.

Reply:
should we put user action in this diagram? if yes, how?